src/main/java/com/example/statussvc/controller/HostsController.java

```diff
@@ -35,8 +49,21 @@ public Object retrieve() {
         return null;
     }
 
-    public Object retrieveAll() {
-        return null;
+    @GetMapping
+    public ResponseEntity<Map<String, Object>> retrieveAll(
+            @RequestParam(defaultValue = "0") int page,
+            @RequestParam(defaultValue = "3") int size
+    ) {
+        Pageable paging = PageRequest.of(page, size);
+        Page<HostRetrieveAllResponse> pageRetrieveAllHostResponses = hostsService.retrieveAll(paging);
+        Map<String, Object> response = new HashMap<>();
+
+        response.put("hosts", pageRetrieveAllHostResponses.getContent());
```

Comment:
you don't need to form peagable here, that's all done by Spring Data Peagable, please check: https://www.baeldung.com/spring-data-jpa-pagination-sorting

Reply:
e.g. just retirn response entity with Page object